src/org/thoughtcrime/securesms/components/reminder/DozeReminder.java

```diff
@@ -110,18 +110,24 @@ public static void dozeReminderTapped(Context context) {
     }
   }
 
-  private static boolean isAllChatmail() {
+  private static boolean isPushAvailableAndSufficient() {
+    if (FcmReceiveService.getToken() == null) {
+      return false;
+    }
+
     for (int accountId : ApplicationContext.dcAccounts.getAll()) {
       DcContext context = ApplicationContext.dcAccounts.getAccount(accountId);
       if (!context.isChatmail()) {
-        return false;
+        // only report "not sufficient" if we're actually connected -
+        // otherwise we'll recheck next time the chatlist is opened
+        int connectivity = context.getConnectivity();
```

Comment:
So, this `isChatmail()` API is really unclear. Maybe the core should just return `true` until it really knows that? Btw, `isChatmail()` affects not only disabling battery optimisation, but also hiding some settings like "move to mvbox". Maybe these settings also should be hidden until the core connects to the server?

Reply:
Yes, but `Context::configure()` doesn't call `inbox_fetch_idle()`. So there's no guarantee that when the user enters Settings, we already know if we a chatmail client. User can do that being offline. Maybe better return `true` until we know?

Reply:
Then `Config::IsChatmail` needs to be set in this block: https://github.com/deltachat/deltachat-core-rust/blob/90c30879b182824f1d823db73bc76e54dc0c4f91/src/configure.rs#L461

Reply:
I'm just trying to understand why `Config::IsChatmail` is set in `inbox_fetch_idle()`, but other chatmail-related config keys here: https://github.com/deltachat/deltachat-core-rust/blob/90c30879b182824f1d823db73bc76e54dc0c4f91/src/configure.rs#L461

Reply:
So, it's guaranteed that `inbox_fetch_idle()` is called by the time `Context::configure()`, and `Config::IsChatmail` is set properly? Then i don't understand why the subject fix needs to look into the connectivity state

Reply:
Ok, not sure my understanding is correct, but anyway i've created a PR for this: https://github.com/deltachat/deltachat-core-rust/pull/5671